cmd/mt-store-cat/main.go

```diff
@@ -228,25 +237,41 @@ func main() {
 	}
 	var metrics []Metric
 	if metricSelector == "*" {
-		fmt.Println("# Looking for ALL metrics")
+		if verbose {
+			fmt.Println("# Looking for ALL metrics")
+		}
 		// chunk-summary doesn't need an explicit listing. it knows if metrics is empty, to query all
 		// but the other two do need an explicit listing.
 		if format == "points" || format == "point-summary" {
-			metrics, err = getMetrics(store, "")
+			metrics, err = getMetrics(store, "", "", "")
 			if err != nil {
 				log.Errorf("cassandra query error. %s", err.Error())
 				return
 			}
 		}
-	} else if strings.HasPrefix(metricSelector, "prefix:") {
-		fmt.Println("# Looking for these metrics:")
-		metrics, err = getMetrics(store, strings.Replace(metricSelector, "prefix:", "", 1))
+	} else if strings.HasPrefix(metricSelector, "prefix:") || strings.HasPrefix(metricSelector, "substr:") || strings.HasPrefix(metricSelector, "glob:") {
+		var prefix, substr, glob string
+		if strings.HasPrefix(metricSelector, "prefix:") {
+			prefix = strings.Replace(metricSelector, "prefix:", "", 1)
+		}
+		if strings.HasPrefix(metricSelector, "substr:") {
+			substr = strings.Replace(metricSelector, "substr:", "", 1)
+		}
+		if strings.HasPrefix(metricSelector, "glob:") {
```

Comment:
This section with so many redundant `strings.HasPrefix()` seems kind of strange to me. Wouldn't it be way cleared if there's simply one case for each of the possible metric selector methods. Then each of those metric selectors could call it's own matching method instead of the current `match(prefix, substr, glob, ...)` which needs to go through all the possible matching methods even though only one of them can ever be used at once. That selector-specific matching method could then simply be passed into `getMetrics()` by name. This might be personal taste, but i think that would be cleaner.